src/input/styles.scss

```diff
@@ -72,6 +109,22 @@
     box-shadow: none;
   }
 
+  &:hover:not(:focus):not(:disabled):not(:read-only) {
```

Comment:
I think there's a _slight_ risk that we're using a different selector for readonly here (`:read-only` vs `.input-readonly` earlier in the file), having consistency would probably be better